homeassistant/package_constraints.txt

```diff
@@ -33,7 +33,7 @@ mutagen==1.46.0
 orjson==3.9.2
 paho-mqtt==1.6.1
 Pillow==10.0.0
-pip>=21.3.1,<23.2
+pip>=21.3.1,<23.3
```

Comment:
We can drop the upper limit. The reason we added this, is because of the legacy resolver which we no longer use.

Reply:
I would recommend to keep it. Pip tends to add new deprecations / removals from release to release. By keeping the upper limit and manually updating it, we can ensure that we don't accidentally break CI.

Just an example, `23.2` added new deprecation notices for non-standard version specifiers.
- #96682

Reply:
🤷 I think we don't need to control that.

For all our images we already pin it ourselves.